sycl/source/detail/event_impl.cpp

```diff
@@ -604,9 +604,8 @@ void event_impl::setSubmissionTime() {
   if (!MIsProfilingEnabled && !MProfilingTagEvent)
     return;
 
-  std::weak_ptr<queue_impl> Queue = isHost() ? MSubmittedQueue : MQueue;
-  if (QueueImplPtr QueuePtr = Queue.lock()) {
-    device_impl &Device = QueuePtr->getDeviceImpl();
+  if (auto Queue = isHost() ? MSubmittedQueue.lock() : MQueue.lock()) {
```

Comment:
To be uniform with changes above, plus avoids creation of a `weak_ptr` copy (which has some overhead: https://godbolt.org/z/YErne5446).